src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Shouldn't this be PUBLIC when pthread is a shared library and PRIVATE when pthread is static? Also wouldn't we want to avoid linking pthread as a static library?